app/test_main.py

```diff
@@ -1 +1,27 @@
-# write your code here
+import pytest
+from unittest import mock
+
+from app.main import can_access_google_page
+
+
+@pytest.mark.parametrize(
+    "valid_url, internet_connection, expected_result",
+    [
+        (True, True, "Accessible"),
+        (True, False, "Not accessible"),
+        (False, True, "Not accessible"),
+        (False, False, "Not accessible"),
+    ]
+)
+def test_can_access_google_page(
+        valid_url: bool,
+        internet_connection: bool,
+        expected_result: str
+) -> None:
+    with mock.patch.multiple(
+            "app.main",
+            valid_google_url=mock.Mock(return_value=valid_url),
+            has_internet_connection=mock.Mock(return_value=internet_connection)
+    ):
+        result = can_access_google_page("https://www.google.com")
+        assert result == expected_result
```

Comment:
There is no need to create an additional variable, you could compare the result of the function directly to the expected result
```suggestion
        assert can_access_google_page("https://www.google.com") == expected_result
```

Do not create redundant variables in the future.